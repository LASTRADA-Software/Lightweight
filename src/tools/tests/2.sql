CREATE TABLE IF NOT EXISTS  "Suppliers" (
    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    "name" VARCHAR(30) NOT NULL
);
CREATE TABLE IF NOT EXISTS  "Account" (
    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    "iban" VARCHAR(30) NOT NULL,
    "supplier_id" BIGINT
);
CREATE TABLE IF NOT EXISTS "AccountHistory" (
    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    "credit_rating" INTEGER NOT NULL,
    "account_id" BIGINT
);
