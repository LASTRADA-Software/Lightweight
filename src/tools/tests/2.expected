// SPDX-License-Identifier: Apache-2.0
#pragma once
#include <Lightweight/DataMapper/DataMapper.hpp>
#include <Lightweight/SqlConnection.hpp>
#include <Lightweight/SqlDataBinder.hpp>
#include <Lightweight/SqlQuery.hpp>
#include <Lightweight/SqlQueryFormatter.hpp>
#include <Lightweight/SqlScopedTraceLogger.hpp>
#include <Lightweight/SqlStatement.hpp>
#include <Lightweight/SqlTransaction.hpp>

struct Account;
struct AccountHistory;
struct Suppliers;

struct Suppliers final
{
    Field<int, PrimaryKey::ServerSideAutoIncrement> id;
    Field<std::string> name;
    HasOneThrough<AccountHistory, Account> accountHistory;
}

struct Account final
{
    Field<int, PrimaryKey::ServerSideAutoIncrement> id;
    Field<std::string> iban;
    BelongsTo<&Suppliers::id> supplier;
};

struct AccountHistory final
{
    Field<int, PrimaryKey::ServerSideAutoIncrement> id;
    Field<int> credit_rating;
    BelongsTo<&Account::id> account;
};
